// Copyright 2016 The Chromium Authors. All rights reserved.
// Use of this source code is govered by a BSD-style
// license that can be found in the LICENSE file or at
// https://developers.google.com/open-source/licenses/bsd

syntax = "proto3";

package monorail;

service Users {
  rpc GetUser (GetUserRequest) returns (User) {}
}

message GetUserRequest {
  string email = 1;
}

message User {
  string email = 1;
  int64 id = 2;
}

service Projects {
  rpc ListProjects (ListProjectsRequest) returns (ListProjectsResponse) {}
  rpc UpdateProjectConfiguredLabels (UpdateProjectConfiguredLabelsRequest) returns (Labels) {}
  rpc PatchProjectConfiguredLabels (PatchProjectConfiguredLabelsRequest) returns (Labels) {}
}

message ListProjectsRequest {
  int32 page_size = 1;
  string page_token = 2;
}

message ListProjectsResponse {
  repeated Project projects = 1;
  string next_page_token = 2;
}

message Project {
  string name = 1;
  string summary = 2;
  string description = 3;
}

message UpdateProjectConfiguredLabelsRequest {
  string project = 1;
  Labels labels = 2;
}

message PatchProjectConfiguredLabelsRequest {
  string project = 1;
  repeated Label added_labels = 2;
  repeated Label removed_labels = 3;
}

message Labels {
  repeated Label labels = 1;
}

message Label {
  string name = 1;
  int64 rank = 2;
}

service Issues {
  rpc CreateIssue (CreateIssueRequest) returns (Issue) {}
  rpc DeleteIssueComment (DeleteIssueCommentRequest) returns (Comment) {}
  rpc ComponentBurndown (ComponentBurndownRequest) returns (Comment) {}
}

message CreateIssueRequest {
  string project = 1;
  Issue issue = 2;
}

message Issue {
  string project = 1;
  int64 issue_number = 2;
  string summary = 3;
  string description = 4;
}

message DeleteIssueCommentRequest {
  string project = 1;
  int64 issue = 2;
  int64 comment = 3;
}

message Comment {
  string content = 1;
  bool deleted = 2;
}

message ComponentBurndownRequest {
  string project = 1;
  string component_path = 2;
}

message IssueCount {
  string datetime = 1;
  int64 count = 2;
}

message ComponentBurndownResponse {
  string project = 1;
  string component_path = 2;
  repeated IssueCount issue_counts = 3;
}
